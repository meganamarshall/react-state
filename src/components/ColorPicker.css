.ColorPicker {
  button {
    margin: '.5em';
    padding: 0.5em
  }

  .red {
    background-color: red;
  }

  .green {
    background-color: green;
  }

  .blue {
    background-color: blue;
  }
}
