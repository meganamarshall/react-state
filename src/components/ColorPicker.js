import React, { PureComponent } from 'react';
import styles from './ColorPicker.css';

export default class ColorPicker extends PureComponent {

  return (
    <section className={styles.ColorPicker}>
      <button className={styles.red}>Red</button>
      <button className={styles.green}>Green</button>
      <button className={styles.blue}>Blue</button>
    </section>
  );
}
