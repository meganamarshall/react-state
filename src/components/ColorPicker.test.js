import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('ColorPicker component', () => {
  it('renders the colorpicker', () => {
    const colors = ['red', 'green', 'blue'];
    const wrapper = shallow(<ColorPicker colors={colors} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('changes div class name to clicked color', () => {
    const colors = ['red', 'green', 'blue'];
    const wrapper = shallow(<ColorPicker colors={colors} />);
    wrapper.setState({ color: 'blue' }, () => {
      expect(wrapper.find('div').at(0).hasClass('blue')).toBeTruthy();
    });
  });
});
