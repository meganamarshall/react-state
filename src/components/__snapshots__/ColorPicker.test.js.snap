// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component renders the colorpicker 1`] = `
<section
  className="ColorPicker"
>
  <Button
    color="red"
    colorHandler={[Function]}
    key="0"
  />
  <Button
    color="green"
    colorHandler={[Function]}
    key="1"
  />
  <Button
    color="blue"
    colorHandler={[Function]}
    key="2"
  />
  <div
    className=""
  />
</section>
`;
